underfs/b2/src/main/java/alluxio/underfs/b2/B2UnderFileSystemFactory.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.underfs.b2;
+
+import alluxio.AlluxioURI;
+import alluxio.conf.PropertyKey;
+import alluxio.underfs.UnderFileSystem;
+import alluxio.underfs.UnderFileSystemConfiguration;
+import alluxio.underfs.UnderFileSystemFactory;
+
+import com.google.common.base.Preconditions;
+import com.google.protobuf.ServiceException;
+
+import java.io.IOException;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Factory for creating {@link B2UnderFileSystem}.
+ */
+@ThreadSafe public class B2UnderFileSystemFactory implements UnderFileSystemFactory {
+  @Override public UnderFileSystem create(String path, UnderFileSystemConfiguration conf) {
+    Preconditions.checkNotNull(path, "path is null");
+    if (checkB2Credentials(conf)) {
+      try {
+
+        return B2UnderFileSystem.createInstance(new AlluxioURI(path), conf);
+      } catch (ServiceException e) {
+        throw new RuntimeException(e);
+      }
+    }
+
+    String err = "B2 Credentials not available, cannot create B2 Under File System.";
+    throw new RuntimeException(new IOException(err));
```

Comment:
Since this is thrown if one of the credential parameters are not set, can we include which parameters are not set in the exception message? That would be helpful for the user know which parameter to define to make this UFS work.

Reply:
Exception message changed to include the required properties.